src/protect/common.py

```diff
@@ -256,6 +258,42 @@ def get_file_from_s3(job, s3_url, encryption_key=None, per_file_encryption=True,
     return filename
 
 
+def get_file_from_url(job, any_url, encryption_key=None, per_file_encryption=True,
+                      write_to_jobstore=True):
+    """
+    Downloads a supplied URL (ftp,http,https) that points to a file. The file
+    is downloaded and a subsequently written to the jobstore and the return value is a the path to
+    the file in the jobstore. If URL link is invalid the function will raise an error.
+
+    :param str any_url: URL for the file
+    :param bool write_to_jobstore: Should the file be written to the job store?
+    """
+
+    work_dir = job.fileStore.getLocalTempDir()
+
+    filename = '/'.join([work_dir, str(uuid.uuid4())])
+    url = any_url
+    parsed_url = urlparse(any_url)
+    try:
+        response = urllib2.urlopen(url)
+
+    except urllib2.HTTPError:
+        if (parsed_url.netloc).startswith('s3') or (parsed_url.netloc).startswith('S3'):
+            job.fileStore.logToMaster("Detected https link is for a encrypted s3 file.")
+            filename = get_file_from_s3(job, any_url, encryption_key=encryption_key,
+                                        per_file_encryption=per_file_encryption,
+                                        write_to_jobstore=False)
```

Comment:
`write_to_jobstore=write_to_jobstore`. get_file_from _s3 should get the same value sent to get_file_from_url